drivers/adc/adc_nrfx_saadc.c

```diff
@@ -104,15 +106,26 @@ struct driver_data {
 	uint8_t active_channel_cnt;
 	void *mem_reg;
 	void *user_buffer;
+#ifdef ADC_DRIVER_USES_INTERNAL_TIMER
+	bool enable_internal_timer;
+#endif
+	bool internal_timer_enabled;
 };
 
 static struct driver_data m_data = {
 	ADC_CONTEXT_INIT_TIMER(m_data, ctx),
 	ADC_CONTEXT_INIT_LOCK(m_data, ctx),
 	ADC_CONTEXT_INIT_SYNC(m_data, ctx),
 	.mem_reg = DMM_DEV_TO_REG(DT_NODELABEL(adc)),
+#ifdef ADC_DRIVER_USES_INTERNAL_TIMER
+	.enable_internal_timer = DT_INST_PROP(0, enable_internal_timer),
```

Comment:
This does not seem like an option we should present to the user, and definately not in the devicetree. If we can use the internal timer, just use it :) Is there some reason not to use it?